### PR TITLE
MI-2220: Button Component

### PR DESCRIPTION
**Pull request by tanisha-agarwal**
### Ticket Link
 [MI-2220](https://brightscout.atlassian.net/browse/MI-2220)

### Description
- Created Button Component

### Screenshots
image
image
image
image
image
image
image
image

### CODE REVIEW

File: src/components/Button/Button.component.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+import {Icon} from '@Components/Icon';
+
+import {ButtonProps} from './Button';
+import {StyledButtonContainer} from './Button.styles';
+
+/**
+ * Display the icon and text in button
+ */
```
Comment (avas27JTG) [inline src/components/Button/Button.component.tsx:10]: use single line comment

File: src/components/Button/Button.component.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+const DisplayContent = ({iconName, iconPosition, children}: ButtonProps) => {
+    // Checks if iconName is undefined or not
+    // if yes then place the icon according to the icon-position given
+    // and return children with icon
+    // else return childer without icon
```
Comment (avas27JTG) [inline src/components/Button/Button.component.tsx:15]: can use multi-line comment here

File: src/components/Button/Button.component.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        <StyledButtonContainer
+            iconPosition={iconPosition}
+            variant={variant}
+            className={`mm-button ${className}`}
+            {...restProps}
+        >
+            <DisplayContent
+                iconName={iconName}
+                iconPosition={iconPosition}
+            >{children}</DisplayContent>
```
Comment (avas27JTG) [inline src/components/Button/Button.component.tsx:68]:
```suggestion
            <DisplayContent
                iconName={iconName}
                iconPosition={iconPosition}
            >
               {children}
            </DisplayContent>
```

File: src/components/Button/Button.d.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+ * Icon Position in Button Component
+ *
+ * start - Icon at left of button text
+ *
+ * end - Icon at right of button text
+ */
+export type IconPositionType = 'start' | 'end';
+
+/**
+ * Type for Button State
```
Comment (avas27JTG) [inline src/components/Button/Button.d.ts:29]:
```suggestion
 * Type for button state
```

File: src/components/Button/Button.component.tsx (hunk 1 above)
Comment (tanisha-agarwal) [reply src/components/Button/Button.component.tsx:10]: The benefit of using multiline comments is that 
In vscode, wherever you use this component you'll be able to see what this component does by just hovering over the component. You won't have to go to the file where this component is written to check fo comments
Screenshot from 2022-10-20 13-23-06

File: src/components/Button/Button.component.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+/**
+ * Display the icon and text in button
+ */
+const DisplayContent = ({iconName, iconPosition, children}: ButtonProps) => {
+    /**
+     * Checks if iconName is undefined or not
+     * if yes then place the icon according to the icon-position given
+     * and return children with icon
+     * else return children without icon
+     */
```
Comment (avas27JTG) [inline src/components/Button/Button.component.tsx:17]:
```suggestion
    /**
     * If "iconName" is present then place the icon according to the given icon position and return children with the icon
     * else return children without an icon
     */
```

File: .storybook/main.js [hunk 6]
```diff
@@ -9,7 +9,8 @@ module.exports = {
     "@storybook/addon-links",
```
Comment (avas27JTG) [inline .storybook/main.js:9]: the last commit says `MI-2220: Changes in package lock` this is misleading as it also contains review fixes, try putting more relevant messages like
`[MI-2220]: Changes in package-lock and review fixes`

**Pull request merged**